src/python/pants/bin/daemon_pants_runner.py

```diff
@@ -153,23 +163,18 @@ def __init__(self, socket, args, env, graph_helper, target_roots, services,
     :param PantsServices services: The PantsServices that are currently running.
     :param str metadata_base_dir: The ProcessManager metadata_base_dir from options.
     :param OptionsBootstrapper options_bootstrapper: An OptionsBootstrapper to reuse.
-    :param Exception deferred_exception: A deferred exception from the daemon's pre-fork context.
-                                         If present, this will be re-raised in the client context.
     """
-    super(DaemonPantsRunner, self).__init__(
-      name=self._make_identity(),
-      metadata_base_dir=metadata_base_dir
-    )
-    self._socket = socket
+    self._name = self._make_identity()
+    self._metadata_base_dir = metadata_base_dir
+    self._maybe_shutdown_socket = maybe_shutdown_socket
     self._args = args
     self._env = env
     self._graph_helper = graph_helper
     self._target_roots = target_roots
     self._services = services
     self._options_bootstrapper = options_bootstrapper
-    self._deferred_exception = deferred_exc
-
-    self._exiter = DaemonExiter(socket)
+    self.exit_code = exit_code
+    self._exiter = DaemonExiter(maybe_shutdown_socket)
 
   def _make_identity(self):
```

Comment:
We may not need this anymore. I'm not sure, but it seems related to `ProcessManager`. If we need it, I suspect it will be for Nailgun communication.

Reply:
added TODO